src/services/vehicle.ts

```diff
@@ -0,0 +1,27 @@
+import { Paginated, SearchType, WithValidationError } from "../index"
+import { ApiCallOptions, fetchPath, handleValidationError } from "../base"
+
+export const fetchFrameNumberTypes = async ({
+  query,
+  options = {},
+}: {
+  query: { dealerId: number; frameNumber: string }
+  options?: ApiCallOptions
+}): Promise<WithValidationError<Paginated<SearchType>>> => {
+  try {
+    const { dealerId, frameNumber } = query
+    const result = await fetchPath({
+      path: `dealers/${dealerId}/vehicles/frame-number/${frameNumber}`,
+      options: {
+        isAuthorizedRequest: true,
+        ...options,
+      },
+    })
+    return {
+      tag: "success",
+      result: { content: result.types, pagination: null },
+    }
+  } catch (error) {
+    return handleValidationError(error)
```

Comment:
How can a fetch request have a validation error/ whats the idea behind wrapping the fetch call into the envelope structure we use for POST and PUT requests?

Reply:
because `frameNumber` will be validated in BE